rosidl_parser/rosidl_parser/__init__.py

```diff
@@ -445,8 +445,7 @@ def parse_value_string(type_, value_string):
     if type_.is_primitive_type() and not type_.is_array:
         return parse_primitive_value_string(type_, value_string)
 
-    # TODO(mikaelarguedas) change this condition once string escape function is implemented
-    if type_.is_primitive_type() and type_.is_array and type_.type != 'string':
+    if type_.is_primitive_type() and type_.is_array:
```

Comment:
Yes sorry this is still work in progress, the strings are not escaped yet. This is also the case for simple strings. e.g. `string def_string 'Hello w"orld!'` makes the message generation crash (at least the python generator).

For string arrays it gets even trickier because the design document says that:
`string:
  a string value which can optionally be quoted with either single quotes (') or double quotes (")`
We need to make the quoting mandatory to allow differentiation of array elements otherwise it'll be ambiguous